### PR TITLE
Fix `test_agent_groups`

### PR DESCRIPTION
**Pull request by juliamagan**
|Related issue|
|-------------|
|      #3865       |

## Description

After the development done in https://github.com/wazuh/wazuh/pull/16050, the behavior of `agent-groups` has changed, so we needed to adapt `test_agent_groups`.

### Added

- Added `execute_wdb_query` in `system/__init__.py`

### Updated

- Updated `test_agent_groups.py`
- Updated `clean_environment()` from `system/conftest.py`
- Updated system tests `README.md`
- Updated `enrollment_cluster` playbook

---

### PR test
| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_agent_groups.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10578928/R3_test_agent_groups.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10578929/R2_test_agent_groups.zip)[🟢 ](https://github.com/wazuh/wazuh-qa/files/10578930/R1_test_agent_groups.zip)       |    9494b96  | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |             | Nothing to highlight |

### Affected tests

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_agent_default_group_added.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10579049/R1_test_agent_default_group_added.zip)          |    ea98e57      | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |            | Nothing to highlight |

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_agent_groups_forced_change.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | ⚫ |             | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |          | Nothing to highlight |

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_agent_groups_new_cluster_node.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | ⚫|              | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |              | Nothing to highlight |

| Tester             | Test path | Jenkins | Local   | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🔴 ](https://github.com/wazuh/wazuh-qa/files/10579054/R1_test_assign_agent_group_with_enrollment.zip) |    ea98e57       | The error is not due to the change affecting this test |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |             | Nothing to highlight |

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_aassign_agent_never_connected_to_group.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10579085/R1_test_assign_agent_never_connected_to_group.zip) |    ea98e57       | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        | Nothing to highlight |

| Tester             | Test path | Jenkins | Local   | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10579120/R1_test_assign_agent_to_a_group_api.zip)|      ea98e57     | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |         | Nothing to highlight |

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_assign_agent_to_a_group_by_tool.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10579204/R1_test_assign_agent_to_a_group_by_tool.zip) |      ea98e57      | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_assign_agent_to_a_group.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🟢 ](https://github.com/wazuh/wazuh-qa/files/10579210/R1_test_assign_agent_to_a_group.zip)|        ea98e57      | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |           | Nothing to highlight |

| Tester             | Test path | Jenkins | Local | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_assign_group_guess.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | [🔴 ](https://github.com/wazuh/wazuh-qa/files/10579389/R1_test_assign_groups_guess.zip)|       ea98e57       | The error is not due to the change affecting this test |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |             | Nothing to highlight |

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_groups_sync_default.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | ⚫ |           | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |            | Nothing to highlight |

| Tester             | Test path | Jenkins | Local  | Commit | Notes                |
|--------------------|-----------|---------|--------|--------|----------------------|
| @juliamagan (Developer)  |     `test_cluster/test_agent_groups/test_groups_sync_time.py`      | :no_entry_sign: :no_entry_sign: :no_entry_sign:  | ⚫ |              | Nothing to highlight |
| @user (Reviewer)   |           | :no_entry_sign: :no_entry_sign: :no_entry_sign: | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |               | Nothing to highlight |

### CODE REVIEW

File: tests/system/test_cluster/test_agent_groups/test_agent_groups.py [hunk 1]
```diff
@@ -122,69 +68,53 @@ def test_agent_groups_create_remove_group(clean_cluster_logs):
...
     # Check if the new information is present in the master and workers dbs
     sleep(time_to_sync)
-    for host in testinfra_hosts:
-        result = host_manager.run_command(host,
-                                          f"{WAZUH_PATH}/framework/python/bin/python3.9 "
-                                          f"{script_path} '{queries[0].format(name=test_group)}'")
-        assert not result, f"This db query should not have returned anything in {host}, but it did: {result}"
+    for host in test_infra_managers:
+        created_group = execute_wdb_query(f"global '{queries[0]}'", host, host_manager)
+        print(created_group)
```
Comment (jmv74211) [inline tests/system/test_cluster/test_agent_groups/test_agent_groups.py:116]: Remove this print

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ Release report: TBD
 
 ### Changed
 
+- Fix `test_agent_groups` ([#3889]([#3769](https://github.com/wazuh/wazuh-qa/pull/3889)) \- (Tests + Framework)
```
[the remark this answers is not in the document]
Comment (juliamagan) [reply CHANGELOG.md]: Done in 3b9b23d

File: tests/system/test_cluster/test_agent_groups/test_agent_groups.py (hunk 1 above)
Comment (juliamagan) [reply tests/system/test_cluster/test_agent_groups/test_agent_groups.py]: Done in 28b2c53

**Review by jmv74211 — APPROVED** [review]
Environment provisioned successfully 

```
ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"wazuh_branch":"fix/16042-remove-groups-sync-task"}'
```

And the tests have passed successfully:

```
test_agent_groups.py::test_agent_groups_create_remove_group PASSED                                                                                                                                          [100%]

========================================================================================== 1 passed in 201.02s (0:03:21) ==========================================================================================
```

**Pull request merged**